src/platform_impl/linux/util.rs

```diff
@@ -27,3 +28,24 @@ pub fn cursor_position(is_wayland: bool) -> Result<PhysicalPosition<f64>, Extern
       .ok_or(ExternalError::Os(os_error!(super::OsError)))?
   }
 }
+
+pub fn is_unity() -> bool {
+  if let Ok(child) = Command::new("echo")
+    .arg("$XDG_CURRENT_DESKTOP")
+    .stdout(Stdio::piped())
+    .spawn()
+  {
+    if let Ok(output) = child.wait_with_output() {
+      if let Ok(string) = String::from_utf8(output.stdout) {
+        let string = string.as_str().to_lowercase();
+        string == "unity" || string == "gnome"
+      } else {
+        false
+      }
+    } else {
+      false
+    }
+  } else {
+    false
+  }
```

Comment:
I think those `false` can be merged into one and return `true` when the string matched

```rust
pub fn is_unity() -> bool {
  if let Ok(child) = Command::new("sh")
    .arg("-c")
    .arg("echo -n $XDG_CURRENT_DESKTOP")
    .stdout(Stdio::piped())
    .spawn()
  {
    if let Ok(output) = child.wait_with_output() {
      if let Ok(string) = String::from_utf8(output.stdout) {
        let string = string.as_str().to_lowercase();
        return string == "unity" || string == "gnome";
      }
    }
  }
  false
}
```

Reply:
Well, idk why it didn't came to my mind...